src/libs/actions/Workflow.ts

```diff
@@ -195,43 +207,135 @@ function removeApprovalWorkflow(policyID: string, approvalWorkflow: ApprovalWork
+ * @param approver - The new approver to set
+ * @param approverIndex - The index of the approver to set
+ * @param policyID - The ID of the policy
+ */
+function setApprovalWorkflowApprover(approver: Approver, approverIndex: number, policyID: string) {
+    const policy = allPolicies?.[`${ONYXKEYS.COLLECTION.POLICY}${policyID}`];
+
+    if (!currentApprovalWorkflow || !policy?.employeeList) {
         return;
     }
 
-    const updatedApprovers = [...currentApprovalWorkflow.approvers];
-    updatedApprovers[index] = approver;
-    Onyx.merge(ONYXKEYS.APPROVAL_WORKFLOW, {approvers: updatedApprovers});
+    const errors: Record<string, TranslationPaths | null> = {};
+
+    // Update the approver at the specified index and reset hints
+    const approvers: Array<Approver | undefined> = currentApprovalWorkflow.approvers.map((existingApprover) => {
+        if (!existingApprover) {
+            return;
+        }
+
+        return {...existingApprover, isInMultipleWorkflows: false};
+    });
+    approvers[approverIndex] = approver;
+
+    // Check if the approver forwards to other approvers and add them to the list
+    if (policy.employeeList[approver.email]?.forwardsTo) {
+        const personalDetailsByEmail = lodashMapKeys(personalDetails, (value, key) => value?.login ?? key);
+        const additionalApprovers = calculateApprovers({employees: policy.employeeList, firstEmail: approver.email, personalDetailsByEmail}).map((additionalApprover) => ({
+            ...additionalApprover,
+            isInMultipleWorkflows: true,
+        }));
+        approvers.splice(approverIndex, approvers.length, ...additionalApprovers);
+    }
+
+    // Check for circular references and reset errors
+    const updatedApprovers = approvers.map((existingApprover, index) => {
+        if (!existingApprover) {
+            return;
+        }
+
+        const hasCircularReference = approvers.slice(0, index).some((previousApprover) => existingApprover.email === previousApprover?.email);
+        if (hasCircularReference) {
+            errors[`approver-${index}`] = 'workflowsPage.approverCircularReference';
+        } else {
+            errors[`approver-${index}`] = null;
+        }
+
+        return {
+            ...existingApprover,
+            isCircularReference: hasCircularReference,
+        };
+    });
+
+    Onyx.merge(ONYXKEYS.APPROVAL_WORKFLOW, {approvers: updatedApprovers, errors});
 }
 
-function setApprovalWorkflow(approvalWorkflow: ApprovalWorkflow) {
-    Onyx.merge(ONYXKEYS.APPROVAL_WORKFLOW, approvalWorkflow);
+function clearApprovalWorkflowApprover(approverIndex: number) {
+    if (!currentApprovalWorkflow) {
+        return;
+    }
+
+    // Update the approver at the specified index and reset hints
+    const approvers: Array<Approver | undefined> = currentApprovalWorkflow.approvers.map((existingApprover) => {
+        if (!existingApprover) {
+            return;
+        }
+
+        return {...existingApprover, isInMultipleWorkflows: false};
```

Comment:
Why do we always reset isInMultipleWorkflows to false? If an approver belongs to three workflow, then we clear this approval flow, this approver still be in two other flows

Reply:
We have to clear hints whenever any approver is updated, because hint messages are based on previous approvers and there is an edge case where a previous approver is removed and its successor hint message is changed which can be confusing. That is why `isInMultipleWorkflows` is always cleared

Reply:
Resolved now in https://github.com/Expensify/App/pull/46798/commits/f554da6696005e996aa1f7aa729d9519b763d108